docs/rfcs/configuration-merging-strategy.md

```diff
@@ -0,0 +1,102 @@
+
+The main motivation for this change was to allow users to define configuration fragments in different sources, and have them merged in such a way that all specified components are included under `service::pipeline` in the final configuration.
+
+Previously, we relied on Koanf’s default merging strategy, which overrides static values and slices (such as strings, numbers, and lists). This behavior often resulted in configurations being unintentionally overwritten when merged from multiple sources.
+
+This issue has been highlighted in several discussions and feature requests:
+- https://github.com/open-telemetry/opentelemetry-collector/issues/8394
+- https://github.com/open-telemetry/opentelemetry-collector/issues/8754
+- https://github.com/open-telemetry/opentelemetry-collector/issues/10370
+
+## Motivation and Scope
+
+We’ve already implemented a feature gate and foundational logic that supports merging lists across configuration files. Currently, this logic is hardcoded to merge lists only for specific keys: receivers, exporters, and extensions. The relevant implementation can be found [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/confmap/merge.go).
+
+The current implementation lacks flexibility. Ideally, users should be able to specify which configuration paths should be merged, rather than relying on hardcoded defaults. 
+This RFC proposes extending the existing functionality by introducing a user-configurable mechanism to define merge behavior.
+
+This RFC builds on top of feedback gathered from the [original PR](https://github.com/open-telemetry/opentelemetry-collector/pull/12097).
+More specifically, this RFC aims to:
+1. Add an option to specify which configuration paths should be merged.
+2. Introduce support for prepend and append operations when merging list values:
+    - This is good to have for lists that rely on certain ordering, such as:
+        - `processors`
+        - `transformprocessor` statements
+
+## Proposed approach
+
+The proposed approach will rely on concept of URI query parameters([_RFC 3986_](https://datatracker.ietf.org/doc/html/rfc3986#page-23)). Our configuration URIs already adhere to this syntax and we can extend it to support query params instead adding new CLI flags. 
+
+For now, the new merging strategy is only enabled under `confmap.enableMergeAppendOption` gate. If user specifies the options and tries to run the collector without gate, we will merge as per default behaviour.
+
+We will support new parameters to config URIs as follows:
+1. `merge_paths`: A comma-separated list of glob patterns which will be used while config merging
+    - This setting will control the paths user wants to merge from the given config.
+    - Example: 
+        - `otelcol --config main.yaml --config extra.yaml?merge_paths=service::extensions,service::**::receivers`
```

Comment:
I think using the query params is fine. However, we need to keep in mind that some confmap providers use the same pattern to pass extra information. With this approach, they won't be able to use these keys which is probably fine. 

If anyone else have any other ideas, please share.

 cc @open-telemetry/collector-approvers

Reply:
We discussed this in [this RFC](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/rfcs/configuring-confmap-providers.md) and came up with a few issues with query parameters that we'll want to consider before going that route.

@dmitryax I believe you, @mx-psi, and I had an offline conversation at KubeCon and tacitly agreed we should go the route of configuring these inside a config file ([this option](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/rfcs/configuring-confmap-providers.md#specify-additional-config-sources-inside-the-main-collector-configuration) in the RFC). Do you still think we should go this route? I think we should probably at least explore it before deciding on an approach.

Reply:
@evan-bradley @dmitryax approach 1 does not deal with params, does this resolve your concerns here?